optimade/validator/validator.py

```diff
@@ -460,16 +465,58 @@ def test_as_type(self):
         self.deserialize_response(response, self.as_type_cls)
 
     @test_case
-    def test_page_limit(self, response):
-        """ Test that a multi-entry endpoint obeys the page limit. """
+    def test_page_limit(self, response, check_next_link=True):
+        """ Test that a multi-entry endpoint obeys the page limit.
+
+        Parameters:
+            response (requests.Response): the response to test for page limit
+                compliance.
+
+        Keyword arguments:
+            check_next (bool): whether or not to recursively follow and test any
+                pagination links provided under `links->next`.
+
+        Raises:
+            ResponseError: if test fails in a predictable way.
+
+        Returns:
+            bool, str: True if the test was successful, with a string describing
+                the success.
+
+        """
         try:
-            num_entries = len(response.json()["data"])
+            response = response.json()
         except AttributeError:
```

Comment:
We also need to import `json` by first trying to import `simplejson as json` and then falling back to importing `json` if `simplejson` is not available.
This is because the `request` library does the same.
```suggestion
        except json.JSONDecodeError:
```

Reply:
OR if you succeed to have `test_case` catch all errors and raise `ResponseError`, or similar, then there is no need to even try/except here.
However, this may result in a loss of granularity - especially considering the excellent rewrite you have done here.

Reply:
I'm pretty sure I've had this raise an `AttributeError` at one point so I'd rather keep protecting against that case too. I've implemented the simplejson/json fallback though.

Reply:
That's true if the `response` is `None` or for the earlier version of this code if `data` was not found in the response. So maybe just add both `Exception`s to the `except` statement? :)